homeassistant/components/bluetooth_le_tracker/device_tracker.py

```diff
@@ -117,13 +128,34 @@ def discover_ble_devices():
     def update_ble(now):
         """Lookup Bluetooth LE devices and update status."""
         devs = discover_ble_devices()
+        adapter = pygatt.GATTToolBackend()
         for mac in devs_to_track:
             if mac not in devs:
                 continue
 
             if devs[mac] is None:
                 devs[mac] = mac
-            see_device(mac, devs[mac])
+
+            battery = None
```

Comment:
@guillempages How will scanning the battery state impact the battery itself?  With this cause additional battery drain? If so it might be best to only do it every X hours

Reply:
It might indeed affect the battery, although I expect it to not be dramatic.

I also had the idea of only checking once a day, but somehow could not find a nice way to do this. That's why I set the default to false, and added a note in the documentation.

Any suggestions how to check that on a device basis? Should I keep an internal dictionary of devices and when they were last checked? I didn't want to change the existing code much, this might need a bigger refactoring.

Reply:
It could be as simple as maintaining a counter that only happens every X CONF_SCAN_INTERVALs

Reply:
I already thought about that, but if devices appear and disappear, there is the possibility that a device is never visible when the battery check is done, so I think a per-device counter would be better.
I did not want to spend much time in this before knowing whether there is interest in accepting the PR; if there would have been a "no way, we don't want that" reaction from the beginning, it would have been wasted effort ;-)

I'll think about it and update the PR. Thanks for the feedback!

Reply:
Ok, I'm tracking the last battery check for each device, and only scanning at most once every 24h. This should be enough (IMHO, if a "BT Low Energy" device uses up the battery noticeably in 24h, it doesn't deserve the "Low Energy" name ;-)), and avoids wasting the battery.

I've tried making the interval configurable, but since I cannot validate the configuration (or at least I didn't find out how), I wasn't able to create a timedelta from the configuration string. Suggestions how to do that are welcome.

I've had the battery check running in my live system for the whole week, with the interval changed to hourly, and did not lose a single percent of the battery (3V CR2032 cell), so battery drain doesn't seem to be that huge an issue.